website/events/services.py

```diff
@@ -151,7 +150,12 @@ def cancel_registration(member, event):
         registration.date_cancelled = timezone.now()
         registration.save()
     else:
-        raise RegistrationError(_("You are not registered for this event."))
+        if registration is not None and registration.payment is not None:
+            raise RegistrationError(
+                _("You can't cancel a registration after having payed.")
+            )
+        else:
+            raise RegistrationError(_("You are not registered for this event."))
```

Comment:
I personally don't really like this approach since I'm basically checking for the same condition again, but I'm not sure either if returning a general error message like "You can't deregister for this event" is ideal.

Reply:
Technically you are creating an if/else inside an else. So it could already be be

```
elif registration and registration.payment:
    ... paid
else:
   ... not registered
```

Reply:
I'm not sure if the error is even used though, so maybe a general one doesn't matter?